public/js/common/react_modal.js

```diff
@@ -0,0 +1,78 @@
+function openReactModal(courseCode) {
+    'use strict';
+    var courseName = getCourseTitle(courseCode);
+    var formattedName = formatCourseName(courseCode);
+    var courseVideoUrls = getCourseVideoUrls(formattedName);
+    
+    React.render(
+        <ModalContent course={formattedName[0]} />,
+        document.getElementById('modal-content-container')
+    );
+}
+
+var ModalContent = React.createClass({
+    render: function() {
+        return (
+            <div>
+                <Description course={this.props.course} />
+            </div>
+        );
+    }
+});
+
+var Description = React.createClass({
+    getInitialState: function() {
+        return {
+            course: [],
+            sessions: []
+        };
+    },
+
+    componentDidMount: function(){
+        //This loads the course json
+        $.ajax({
+            url: 'course',
+            data: {name: this.props.course},
+            dataType: 'json',
+            success: function(data) {
+                if (this.isMounted()) {
+                    this.setState({course: data});
+                    //Not sure if there is a cleaner way to do this
+                    //This is getting the session times
+                    var lectures = data.fallSession.lectures
+                                                   .concat(data.springSession.lectures)
+                                                   .concat(data.yearSession.lectures);
+                    var sessions = [];
+                    $.each(lectures, function( index, value ) {
```

Comment:
Use the Array prototype `map` function (something like `lectures.map( function ...`) to return an array.